app/controllers/application_drafts_controller.rb

```diff
@@ -63,7 +63,7 @@ def apply
     if current_team.coaches_confirmed?
       if application_draft.ready? && application_draft.submit_application!
         flash[:notice] = 'Your application has been submitted!'
-        ApplicationFormMailer.new_application(application_draft.application).deliver_later
+        application_draft.application.notify_orga_and_submitters
```

Comment:
(ugh, for some reason that comment didn't make it into my review – PEBCAC for sure)

Personal taste: I tend to `!` methods with IO/network side effects. Then again, I also like it that the method doesn't say that the type of notification is an email. /shrug :) 

One other thing, though: can we move this notification into the state transition? The method call on `application` looked a bit suspicious (Mrs Demeter called) and that got me thinking that it's probably not the controller's responsibility

Reply:
Haha, nice that we're on the same page here @carpodaster !

I actually had both, the `ApplicationDraft` model's state machine and the `CreateApplicationEtc...` service refactored to do exactly this 😹 

But I had to adjust too many tests and did not have the time to finish this on Sunday. So I'll do it in another PR.

Btw., a thing I changed there: I'd prefer the `CreateApplicationEtc.` service to actually **return** the persisted application record to then use this instance for notifying orga and the submitters. Using `deliver_later` for the emails requires all DB transactions for the `draft` and the `application` to be completed - assigning the `application` makes this a bit more explicit to me ✌️

Reply:
So, are you fine with "fixing" this in a follow up @carpodaster ?